### ISSUE TITLE
General way to handle groupId in IDisplayManager.UpdateEditorAsync

### ISSUE THREAD

**Issue by alberthajdu** [OrchardCMS/OrchardCore#1834]
When you tell a driver to display it's shape only in case of a specific group: `OnGroup("groupId")` for `DisplayDriver.Display` or `DisplayDriver.Edit` then it triggers only those drivers that meant to that group.

But when you want to trigger only those driver's `DisplayDriver.UpdateAsync` that are on that specific group then the drivers without groups triggered as well.

My use-case would be adding a new editor for the User entity: my DisplayDriver.Edit displays my shape with the help of the `.OnGroup("myGroupId")` but when I try to save it and update it with `await _userDisplayManager.UpdateEditorAsync((User)user, this, false, "myGroupId");` the default driver's (`UserDisplayDriver : DisplayDriver<User>`) `DisplayDriver.UpdateAsync` triggers too and produces validation errors because the UserName and Email are required on the `EditUserViewModel`.

I've checked how ISite handles this stuff and I noticed that it works only because the base entitiy driver `DefaultSiteSettingsDisplayDriver` has a default group id and performs a check `if (context.GroupId == GroupId)` before updates its model. And its extensions like `RegistrationSettingsDisplayDriver ` performs their checks on their groupIds too.

So it's not consistent and in case of an Entity author forgets this little check and a model contains required properties then who wants to extend that Entity faces a wall. Like with the User Entity.

I don't know how to solve this problem and maybe it was a design decision. Can someone please explain the reasons behind this?

**Comment by sebastienros** [comment]
I think I am facing this issue on the code I am currently writing. Once I find a solution I will comment here.

**Comment by sebastienros** [comment]
If your update method, are you using `Prefix` when updating the model?

**Comment by alberthajdu** [comment]
Yes, I'm using it: `await context.Updater.TryUpdateModelAsync(model, Prefix);` 
Do you need any more details?

**Comment by sebastienros** [comment]
Can you try to test the `context.GroupId` value in the Update method? 

Something https://github.com/OrchardCMS/OrchardCore/blob/dev/src/OrchardCore.Modules/OrchardCore.Users/Drivers/RegistrationSettingsDisplayDriver.cs#L25

**Comment by alberthajdu** [comment]
I don't understand, why and how should I test that?

Grouping in RegistrationSettingsDisplayDriver works but only because the default driver for ISite https://github.com/OrchardCMS/OrchardCore/blob/dev/src/OrchardCore.Modules/OrchardCore.Settings/Drivers/DefaultSiteSettingsDisplayDriver.cs#L28 uses a default GroupId. But https://github.com/OrchardCMS/OrchardCore/blob/dev/src/OrchardCore.Modules/OrchardCore.Users/Drivers/UserDisplayDriver.cs#L70 doesn't have a default GroupId so it will run always. My issue would be resolved if UserDisplayDriver would use a default GroupId but maybe that's not the perfect solution because that's implicit and Entity authors will forget that.
